src/java/picard/sam/markduplicates/util/ReadEndsForMarkDuplicates.java

```diff
@@ -71,4 +71,9 @@ public ReadEndsForMarkDuplicates(final ReadEndsForMarkDuplicates read) {
         this.read1IndexInFile = read.read1IndexInFile;
         this.read2IndexInFile = read.read2IndexInFile;
     }
+
+    @Override
+    public ReadEndsForMarkDuplicates clone() {
```

Comment:
That was my initial preference as well but I couldn't make it work. the problem is that I do not know the actual type of the object since it could be either a ReadEndsForMarkDuplicates or a ReadEndsForMarkDuplicatesWithBarcodes. if there's a way to do it without a clone, let me know.